src/shims/fs.rs

```diff
@@ -257,6 +264,196 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         this.try_unwrap_io_result(result)
     }
 
+    fn statx(
+        &mut self,
+        dirfd_op: OpTy<'tcx, Tag>,
+        pathname_op: OpTy<'tcx, Tag>,
+        flags_op: OpTy<'tcx, Tag>,
+        _mask_op: OpTy<'tcx, Tag>,
+        statxbuf_op: OpTy<'tcx, Tag>
+    ) -> InterpResult<'tcx, i32> {
+        let this = self.eval_context_mut();
+
+        this.check_no_isolation("statx")?;
+
+        let statxbuf_scalar = this.read_scalar(statxbuf_op)?.not_undef()?;
+        let pathname_scalar = this.read_scalar(pathname_op)?.not_undef()?;
+
+        // If the statxbuf or pathname pointers are null, the function fails with `EFAULT`.
+        if this.is_null(statxbuf_scalar)? || this.is_null(pathname_scalar)? {
+            let efault = this.eval_libc("EFAULT")?;
+            this.set_last_error(efault)?;
+            return Ok(-1);
+        }
+
+        // If statxbuf is not null we need to produce a proper `MemPlace` to write the results of
+        // the function to it. This is needed because all the `statx` parameters are provided with
+        // an `i64` layout instead of the proper type layout.
+        let statxbuf_ptr = this.force_ptr(statxbuf_scalar)?;
```

Comment:
`force_ptr` is almost always wrong and you want `check_ptr_access` or `check_mplace_access` instead.

Also I don't understand the comment, and it seems to have nothing to do with the code. The code creates a `Pointer`, but the comment talks about a `MemPlace` which doesn't even need a `Pointer`.

Reply:
Oh I think what you want to say is that `statxbuf_op` is not the type you want it to have? Maybe? So does `statxbuf_op` have type `i64` where it should be `*mut statx`? But that makes little sense; it should be `isize` at least...

Please explain what that comment means and where there is an `i64`, as I don't get it. :)

Reply:
That was me believing that every platform has a 64 bit architecture :P 

I meant what you said in the last comment. `statxbuf` is a raw integer instead of being a proper pointer to `statx`. I'll rewrite it to make it clearer.